### PR TITLE
BIP32/39 sign up

### PR DESCRIPTION
**Pull request by alexanderdushkin**
This pull request implements BIP32/39 sign up with Rein. Both web-based and CLI signup, as well as account import have been rewritten. Particularly, Rein now signs all enrollment documents automatically.

CAUTION: This update would probably break backwards compatibility with older Rein local databases. As requested by @weex I added the delegate private master key to the identity table in the DB, so the user model has changed. I couldn't test the thing, but it should fail as no `dxprv` field is present in old databases.

This is solved by deleting your local Rein data with `rm -rf ~/.rein` and importing your data from backup. The import process has been designed to sign your enrollment document with your mnemonic if you used the new version or your master private key if you used the old version. In the latter case the `dxprv` field is set to `null` in the DB. As per issue #76, Rein does not create duplicates when you request microhosting space so you can safely do `rein request <server>` and then `rein sync` while preserving your account.

### CODE REVIEW

**Comment by weex** [comment]
This is really cool @adsalpha, looking forward to further testing on this before integrating.

**Comment by alexanderdushkin** [comment]
@weex great, just lmk if you need me to test something specifically.

**Comment by weex** [comment]
Gave this a bit of testing and though the signup and and request work I can't seem to get to any pages after setup. /, /bid, /post etc say not found. I tested this on a vm I had setup for other testing so not sure if there's some conflict. Just want to verify if this expected behavior for now.

**Comment by alexanderdushkin** [comment]
Not really expected, although I touched neither of those pages. Have you used `rein request` and `rein sync` prior to accessing those URIs? If you did, lmk I will review the issue.

I only worked on my module and for the server, everything remains the same. At least that's what I expected, that's why I only tested my module.

**Comment by weex** [comment]
The issue of not getting into any pages was because the project didn't have a MANIFEST.in file. This is why installing with `pip install --editable .` worked but `python setup.py install` did not. The latter needs to know which non .py files to copy whereas the first just links the OS directly to the source directory which includes rein/html.*.

One other question though @adsalpha. Can you point to the bip32 path scheme this is using and where that lives in the code? Thanks.

**Comment by alexanderdushkin** [comment]
I think I will add it to 'Addresses and payments' in the README file.

File: rein/cli.py [hunk 1]
```diff
@@ -1185,87 +1174,46 @@ def flash_errors(form):
...
-                try:
-                    f.write(json.dumps(data))
-                    click.echo("Backup saved successfully to %s" % filename)
-                except:
-                    RuntimeError('Problem writing user details to json backup file.')
-                    return render_template("setup.html",
-                                           form=form)
-                f.close()
+    @app.route('/setup')
+    def setup():
```
Comment (weex) [inline rein/cli.py]: Found an issue getting past setup. This function name conflicts silently with the setup argument to the `rein start` command. Please update the function name and this issue should be resolved.

**Comment by alexanderdushkin** [comment]
@weex, done.

**Comment by weex** [comment]
Awesome, thanks.

**Comment by weex** [comment]
Just an FYI, akrmn is integrating this with his latest changes in preparation for a v0.3 release. So this PR won't be merged directly though commits will be included with appropriate attribution.

**Pull request closed without merging**